invenio_rdm_records/resources/resources.py

```diff
@@ -227,3 +241,167 @@ def search(self):
             identity=g.identity,
         )
         return items.to_dict(), 200
+
+
+# IIIF decorators
+
+iiif_request_view_args = request_parser(
+    from_conf("request_view_args"), location="view_args"
+)
+
+
+def with_iiif_content_negotiation(serializer):
+    """Always response as JSON LD regardless of the request type."""
+    return with_content_negotiation(
+        response_handlers={
+            "application/ld+json": ResponseHandler(serializer()),
+        },
+        default_accept_mimetype="application/ld+json",
+    )
+
+
+class IIIFResource(ErrorHandlersMixin, Resource):
+    """IIIF resource."""
+
+    def __init__(self, config, service):
+        """Constructor."""
+        super().__init__(config)
+        self.service = service
+
+    def create_url_rules(self):
+        """Create the URL rules for the IIIF resource."""
+        routes = self.config.routes
+        return [
+            route("GET", routes["manifest"], self.manifest),
+            route("GET", routes["sequence"], self.sequence),
+            route("GET", routes["canvas"], self.canvas),
+            route("GET", routes["image_base"], self.base),
+            route("GET", routes["image_info"], self.info),
+            route("GET", routes["image_api"], self.image_api),
+        ]
+
+    def _get_record_with_files(self):
+        uuid = resource_requestctx.view_args["uuid"]
+        return self.service.read_record(uuid=uuid, identity=g.identity)
+
+    #
+    # IIIF Manifest - not all clients support content-negotiation so we need a
+    # full endpoint.
+    #
+    # See https://iiif.io/api/presentation/2.1/#responses on
+    # "Access-Control-Allow-Origin: *"
+    #
+    @cross_origin(origin="*", methods=["GET"])
```

Comment:
Question: @dfdan What was the reason behind allowing cross origin here? Mirador should be loaded from the same domain so it should be able to access the API?

Reply:
I believe the reason is to allow external "viewers" to load the content, i.e. mirador running on a different website.